pkgs/applications/editors/emacs/default.nix

```diff
@@ -31,7 +32,8 @@ stdenv.mkDerivation rec {
     sha256 = "0cwgyiyymnx4xdg99dm2drfxcyhy2jmyf0rkr9fwj9mwwf77kwhr";
   };
 
-  patches = lib.optional stdenv.isDarwin ./at-fdcwd.patch;
+  patches = lib.optional stdenv.isDarwin ./at-fdcwd.patch
+    ++ lib.optional withCsrc ./install-C-src.patch;
```

Comment:
@binarin I just remembered one advantage of the patch. It is a lot more self contained. Without that you need to pass some target to make and do things in `postInstall`. It sprinkles the option around a little bit, but not that bad.